react/features/base/responsive-ui/middleware.native.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+
+import { Dimensions } from 'react-native';
+
+import { APP_WILL_MOUNT, APP_WILL_UNMOUNT } from '../../base/app';
+import { MiddlewareRegistry } from '../../base/redux';
+
+import { setAspectRatio, setReducedUI } from './actions';
+
+/**
+ * Dimensions change handler.
+ */
+let handler;
+
+/**
+ * Middleware that handles widnow dimension changes and updates the aspect ratio and
+ * reduced UI modes accordingly.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    const result = next(action);
+
+    switch (action.type) {
+    case APP_WILL_UNMOUNT: {
+        _appWillUnmount();
+        break;
+    }
+    case APP_WILL_MOUNT:
+        _appWillMount(store);
+        break;
+
+    }
+
+    return result;
+});
+
+/**
+ * Notifies this feature that the action {@link APP_WILL_MOUNT} is being
+ * dispatched within a specific redux {@code store}.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @private
+ * @returns {void}
+ */
+function _appWillMount(store) {
+    handler = dim => {
+        _onDimensionsChange(dim, store);
+    };
+
+    Dimensions.addEventListener('change', handler);
+}
+
+/**
+ * Notifies this feature that the action {@link APP_WILL_UNMOUNT} is being
+ * dispatched within a specific redux {@code store}.
+ *
+ * @private
+ * @returns {void}
+ */
+function _appWillUnmount() {
+    Dimensions.removeEventListener('change', handler);
+
+    handler = undefined;
+}
+
+/**
+ * Handles window dimension changes.
+ *
+ * @param {Object} dimensions - The new dimensions.
+ * @param {Store} store - The redux store.
+ * @private
+ * @returns {void}
+ */
+function _onDimensionsChange(dimensions, store) {
+    const { width, height } = dimensions.window;
+    const { dispatch } = store;
+
+    dispatch(setAspectRatio(width, height));
+    dispatch(setReducedUI(width, height));
```

Comment:
Do setAspectRatio and setReducedUI always get called together with the same values? Is it worth combining into one action, assuming what they do are very related?

Reply:
They do use the same values, but they compute different things so I thought it made sense to keep them separate. I don’t event know what name I’d give that thing... WDTY?

Reply:
What i'd do is to only dispatch dimensions change event here, and let the appropriate features handle them, meaning the reduced UI should listen to dimension change, not the dimension detector inform the reduced UI to do something.

Reply:
All those 3 things are in the same "umbrella" feature, this one. Dispatching a dimensions changed just to catch it on the same middleware to then do these 2 things felt unnecessary so I took this shortcut. If any other entity ever needs to be aware of the specific dimensions (I hope not, that would be kinda bad) we can change it.